src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8242,24 +8242,14 @@ internal static List<AlternateStreamData> GetStreams(string path)
         /// <returns>A FileStream that can be used to interact with the file.</returns>
         internal static FileStream CreateFileStream(string path, string streamName, FileMode mode, FileAccess access, FileShare share)
         {
-            if (path == null) throw new ArgumentNullException("path");
-            if (streamName == null) throw new ArgumentNullException("streamName");
-
-            string adjustedStreamName = streamName.Trim();
-            adjustedStreamName = ":" + adjustedStreamName;
-            string resultPath = path + adjustedStreamName;
-
-            if (mode == FileMode.Append) mode = FileMode.OpenOrCreate;
-            SafeFileHandle handle = NativeMethods.CreateFile(resultPath, access, share, IntPtr.Zero, mode, 0, IntPtr.Zero);
-
-            if (handle.IsInvalid)
+            if (!TryCreateFileStream(path, streamName, mode, access, share, out var stream))
             {
                 string errorMessage = StringUtil.Format(
                     FileSystemProviderStrings.AlternateDataStreamNotFound, streamName, path);
-                throw new FileNotFoundException(errorMessage, resultPath);
+                throw new FileNotFoundException(errorMessage, $"{path}:{streamName.Trim()}");
```

Comment:
Do we really need `Trim()` for `streamName` in the _internal_ methods? We do not this for `path`.

Reply:
Not sure - I just opted to not change any functionality. Small cost on the error path.

Reply:
I vote to remove - I believe consumer should take care.